src/services/goToDefinition.ts

```diff
@@ -710,3 +738,25 @@ function isConstructorLike(node: Node): boolean {
             return false;
     }
 }
+
+function createDefinitionInfoFromStatement(
+    statement: SwitchStatement,
+    keywordKind: KeywordSyntaxKind,
+    sourceFile: SourceFile,
+    name: string,
+): DefinitionInfo {
+    const keyword = find(statement.getChildren(sourceFile), node => node.kind === keywordKind)!;
+    return {
+        fileName: sourceFile.fileName,
+        textSpan: createTextSpanFromNode(keyword, sourceFile),
+        kind: ScriptElementKind.keyword,
+        name,
+        containerKind: undefined!,
+        containerName: "",
+        contextSpan: createTextSpanFromBounds(keyword.getStart(sourceFile), statement.caseBlock.getFullStart()),

```

Comment:
This needs to be moved to `getContextNode` which handled `SwitchStatement`then `...FindAllReferences.getContextSpan()` so that all places where we will use context span are in single place.

Reply:
Looking at createDefinitionInfoFromName above, it looks like the right way to use this is first to call getContextNode to get the start/end tokens, then to use toContextSpan to turn it into a context span (see code below). Unfortunately, my `getFullStart` hack (which grabs the closing paren) gets lost since getContextNode returns a node pair, not a location pair. And toContextSpan currently always calls getEnd, but my hack uses getFullStart.

```ts
    const keyword = FindAllReferences.getContextNode(statement)!;
    const textSpan = createTextSpanFromNode(isContextWithStartAndEndNode(keyword) ? keyword.start : keyword, sourceFile)
    return {
        fileName: sourceFile.fileName,
        textSpan,
        kind: ScriptElementKind.keyword,
        name: "switch",
        containerKind: undefined!,
        containerName: "",
        ...FindAllReferences.toContextSpan(textSpan, sourceFile, keyword),
        isLocal: true,
        isAmbient: false,
        unverified: false,
        failedAliasResolution: undefined,
    };
```

Any ideas? I'm going to try hacking toContextSpan to special case switches to see how bad that is.

Reply:
OK, the hack isn't too bad; there's a special case for strin gliterals there already, so I added it after that.